projects/jdwp/defs/command_sets/event_request.py

```diff
@@ -0,0 +1,130 @@
+from projects.jdwp.defs.schema import (
+    Command,
+    Field,
+    Struct,
+    CommandSet,
+    IntegralType,
+    Type,
+)
+from projects.jdwp.defs.constants import ErrorType
+from typing import Union
+
+
+CountModifier = Struct([
+    Field("count", IntegralType.INT, "Count before event. One for one-off.")
+])
+
+ConditionalModifier = Struct([
+    Field("exprID", IntegralType.INT, "For the future")
+])
+
+ThreadOnlyModifier = Struct([
+    Field("thread", Type.THREAD_ID, "Required thread")
+])
+
+ClassOnlyModifier = Struct([
+    Field("class_", Type.CLASS_ID, "Required reference type")
+])
+
+ClassMatchModifier = Struct([
+    Field("classPattern", Type.STRING, "Restricted regular expression")
+])
+
+StepModifier = Struct([
+    Field("thread", Type.THREAD_ID, "Required thread"),
+    Field("size", IntegralType.INT, "Size of each step"),
+    Field("depth", IntegralType.INT, "Relative call stack limit")
+])
+
+
+ClassExcludeModifier = Struct([
+    Field("classPattern", Type.STRING, "Restricted regular expression")
+])
+
+LocationOnlyModifier = Struct([
+    Field("loc", Type.LOCATION, "Required location")
+])
+
+ExceptionOnlyModifier = Struct([
+    Field("exceptionOrNull", Type.CLASS_ID, "Optional reference type"),
+    Field("caught", Type.BOOLEAN, "True if exception was caught"),
+    Field("uncaught", Type.BOOLEAN, "True if exception was uncaught")
+])
+
+FieldOnlyModifier = Struct([
+    Field("declaring", Type.CLASS_ID, "Required reference type"),
+    Field("fieldID", IntegralType.INT, "Required field ID")
+])
+
+InstanceOnlyModifier = Struct([
+    Field("instance", Type.OBJECT_ID, "Required object")
+])
+
+SourceNameMatchModifier = Struct([
+    Field("sourceNamePattern", Type.STRING, "Restricted regular expression")
+])
+
+
+__Modifier = Union[
```

Comment:
We want to use `TaggedUnion` type from schema.py to represent the union. This should look something like:

```
__SetCommand_outArray_Element_modKind = Field(
    "modKind",
    UnionTag(IntegralType.BYTE, ModifierKind),
    "Modifier kind",
)

__SetCommand_outArray_Element_Modifier = TaggedUnion(
    tag=__SetCommand_outArray_Element_modKind,
    cases={
        ModifierKind.COUNT: CountModifier,
        ModifierKind.CONDITIONAL: ConditionalModifier,
        ...
    },
)
```

In another PR we will want to enforce that the `cases` dict is total, for every possible enum value it has a corresponding Struct value. It can be done via a unit test :)